CMakeLists.txt

```diff
@@ -0,0 +1,255 @@
+# SPDX-License-Identifier: Unlicense
+#
+# LibTomMath, a free open source portable number theoretic multiple-precision
+# integer (MPI) library written entirely in C.
+#
+
+cmake_minimum_required(VERSION 3.7)
+project(tommath VERSION 1.2.0)
+
+#-----------------------------------------------------------------------------
+# Include cmake modules
+#-----------------------------------------------------------------------------
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(sources.cmake)
+
+# The only direct cmake argument for now
+option(BUILD_SHARED_LIBS "Build shared library and only the shared library if \"ON\", default is static" OFF)
+
+#-----------------------------------------------------------------------------
+# Compose CFLAGS
+#-----------------------------------------------------------------------------
+
+# Some information copied from makefile_include.mk
+
+# Basic set
+set(LTM_C_FLAGS  -Wall -Wsign-compare -Wextra -Wshadow)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wdeclaration-after-statement -Wbad-function-cast -Wcast-align)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wstrict-prototypes -Wpointer-arith)
+
+# Do we run the sanitizer?
+if(DEFINED ENV{SANITIZER})
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -fsanitize=undefined -fno-sanitize-recover=all -fno-sanitize=float-divide-by-zero)
+endif()
+
+if(DEFINED ENV{CONV_WARNINGS})
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -std=c89 -Wconversion -Wsign-conversion)
+    if($ENV{CONV_WARNINGS} EQUAL "strict")
+        set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wc++-compat)
+    endif()
+else()
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wsystem-headers)
+endif()
```

Comment:
Not sure if those are really required here, as this exists only for the unit tests and is already done in the makefiles.

Reply:
Uh, yes, the unit tests use the makefiles, so these lines can go.